srcpkgs/mattermost-desktop/template

```diff
@@ -0,0 +1,50 @@
+# Template file for 'mattermost-desktop'
+pkgname=mattermost-desktop
+version=4.5.2
+revision=1
+archs="x86_64"  # no musl due to electron
+wrksrc="mattermost-desktop-${version}"
+# electron-builder needs GNU tar
+hostmakedepends="git wget nodejs-lts-10 python libicns GraphicsMagick xz tar"
+short_desc="Team messaging app, an open source Slack alternative"
+maintainer="Artem Zhurikhin <ashpool@xecut.net>"
+license="Apache-2.0"
+homepage="https://mattermost.com/"
+checksum=17e32346592d175f091a4f12068e0cfc5cfd20ed7894f88bb5b54b2eb6eac74a
+
+# In the source archive the top-level dir is named "desktop-${version}/"
+# It does not contain the name of the package, so we need custom fetch & extract
+
+_disturl="https://github.com/mattermost/desktop/archive/v${version}.tar.gz"
+_distfile="mattermost-desktop-${version}.tar.gz"
+
+do_fetch() {
+	wget -O "${_distfile}" "${_disturl}"
+}
+
+do_extract() {
+	pwd
+	bsdtar -C ${wrksrc} \
+		-xzf "${XBPS_BUILDDIR}/${_distfile}" \
+		--strip-components=1 \
```

Comment:
Ah, this makes sense now... Just use `wrksrc="desktop-${version}"` in the template and you can let `xbps-src` manage the rest.

Reply:
Ok, but it felt wrong that the `wrksrc` would not contain the name of the package at all. I thought that could cause confusion and collisions. Should I not be concerned with the possibility of another package extracting itself into a directory with such a name?

Reply:
The `$wrksrc` is removed again once the package is built. The name can be arbitrary and doesn't matter most of the time (pun intended :+1: